tests/groups/test_data.py

```diff
@@ -144,31 +143,31 @@ async def test_update_permissions(
             **generate_base_permissions(),
             "create_sample": True,
             "modify_subtraction": True,
-        }
+        },
     )
     assert await get_row_by_id(pg, SQLGroup, group.id) == snapshot(name="pg_added")
 
     group = await data_layer.groups.update(
-        group.id, UpdateGroupRequest(**{"permissions": {"create_sample": False}})
+        group.id, UpdateGroupRequest(permissions={"create_sample": False}),
     )
     assert group == snapshot(name="group_removed")
     assert group.permissions == Permissions(
-        **{**generate_base_permissions(), "modify_subtraction": True}
+        **{**generate_base_permissions(), "modify_subtraction": True},
     )
     assert await get_row_by_id(pg, SQLGroup, group.id) == snapshot(name="pg_removed")
 
 
 class TestDelete:
-    async def test_ok(self, data_layer: DataLayer, pg: AsyncEngine, fake2):
+    async def test_ok(self, data_layer: DataLayer, pg: AsyncEngine, fake):
```

Comment:
Please add typing where it is missing, in files you touch.
```suggestion
    async def test_ok(self, data_layer: DataLayer, pg: AsyncEngine, fake: DataFaker):
```

Reply:
Trying to do this elegantly is taking a while. Pycharm refactor only takes care of rename. No way to ensure import of type in all files.

Might need to manual it in the morning. 
Or get better at grep and bash

Reply:
Don't worry about it. Best effort when you see it while editing. 

Let's get this merged